File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for(int i = 0 ; i<nums.length; i++){
            int req = target - nums[i];
            if(map.containsKey(req)){
                int[] ans = new int[2];
                ans[0] = map.get(req);
                ans[1] = i;
                return ans;
            }
            map.put(nums[i],i);
        }
        return new int[]{};
    }
}

File: 1-two-sum/two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = HashMap<Int,Int>()
        val ans = IntArray(2)
        for(i in 0 until nums.size ){
            map[nums[i]] = i 
        }
    
        for(i in 0 until nums.size ){
            val req = target - nums[i]
            if(map.containsKey(req)){
                if(map.get(req)!=i){
                    ans[0] = i
                    ans[1] = map[req]!!
                    return ans
                }
            } 
        }
        return ans
    }
}